Userland/Libraries/LibLine/Editor.cpp

```diff
@@ -2006,6 +2006,40 @@ VTState actual_rendered_string_length_step(StringMetrics& metrics, size_t index,
     return state;
 }
 
+static Result<char, Editor::Error> try_read_char_for_vt_dsr_with_retries()
+{
+    char c;
+    Editor::Error error;
+    bool read_success = false;
+
+    // Sometimes you have to be a little patient for terminal to start outputting
+    // the response for "Report cursor position". For example, this happens after
+    // Ctrl+C'ing the "top" command.
```

Comment:
Now, for the "real" review:

This seems odd, we come here if `select()` says there _is_ stuff to read, so if we read nothing...what happened?
If trying a couple times in quick succession fixes the problem, it's certainly not a timing issue as the window here is fairly small.